Feature/core 541 permissions ui

karladler opened the pull request:
**Update: can be merged yet, after approve!**

karladler committed:
add $route interface to evan component, allow json loading in ts

- [CORE-541]

karladler committed:
add permissions component

- [CORE-541] add permissions component with a set of dummydata

karladler committed:
add checkbox component

- [CORE-541] add checkbox component and adjust form control component for checkbox placement

karladler committed:
Merge branch 'develop' into feature/CORE-541-permissions-UI

karladler committed:
fix checkbox behavior

karladler committed:
add proper logic for permissions component

- [CORE-541] fix the logic for setting read, when readWrite option is set and unsetting write, when read is unset.

karladler committed:
add permissions editor component

- [CORE-541] add component for permissions editing.

karladler committed:
Merge branch 'develop' into feature/CORE-541-permissions-UI

karladler committed:
update versions.md

- [CORE-541] update versions md

karladler committed:
finalize permissions editor

- [CORE-541] prepare for load / update permissions functions as properties
- fix code styles
- fix tslint.json
- use dummy data from outside / remove from here
- update translations

wulfraem commented on dapps/evancore.vue.libs/src/i18n/de.ts:
```diff
@@ -107,16 +107,25 @@ export default {
+      "selectContact": "Kontakt für die Freigabe auswählen",
+      "defineFor": "Setzen Sie die Zugriffsrechte für ${contactName}.",
+      "permissionsTitle": "Zugriffsrechte",
+      "noContacts": "Sie haben keine Kontakte in ihrem Adressbuch, bitte fügen Sie zuerst einen Kontakt hinzu."
```
```suggestion
      "noContacts": "Sie haben keine Kontakte in Ihrem Adressbuch, bitte fügen Sie zuerst einen Kontakt hinzu."
```

wulfraem reviewed (changes requested):
^^

karladler committed:
Update dapps/evancore.vue.libs/src/i18n/de.ts

fix typo

Co-Authored-By: wulfraem <wulfraem@users.noreply.github.com>

DanielMaass commented on dapps/evancore.vue.libs/src/components/permissions-editor/permissions-editor.ts:
```diff
@@ -0,0 +1,166 @@
+
+import { Prop } from 'vue-property-decorator';
+
+interface PermissionsInterface {
```
interface should define in one place. Maybe in permissions.ts

DanielMaass commented on dapps/evancore.vue.libs/src/components/permissions-editor/permissions-editor.vue:
```diff
@@ -0,0 +1,65 @@
+    <p>
+      To share your profile information with another contact, fill out the form below and click on
+      “Share Profile Data”
+    </p>
```
we should support localization

DanielMaass commented on dapps/evancore.vue.libs/src/components/permissions/permissions.ts:
```diff
@@ -0,0 +1,132 @@
+  [property: string]: {
+    read: boolean,
+    readWrite: boolean,
+    fields: string[]
```
maybe define as optional "fields?: string[]"

DanielMaass commented on dapps/evancore.vue.libs/src/components/permissions/permissions.ts:
```diff
@@ -0,0 +1,132 @@
+   * The dataset name to be displayed.
+   */
+  @Prop({
+    default: ''
```
set as required as well

DanielMaass commented on dapps/evancore.vue.libs/src/components/permissions/permissions.vue:
```diff
@@ -0,0 +1,62 @@
+      </tbody>
+    </table>
+    <div v-else>
+      <p class="p-6 text-error">No permisisons available to edit.</p>
```
support localization

karladler committed:
PR fixes

- move language files to json
- move common interfaces to interfeces.ts
- small PR fixes
- fox tslint.json for semicolons
